plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/RefreshVersionsTask.kt

```diff
@@ -28,6 +29,21 @@ open class RefreshVersionsTask : DefaultTask() {
 
         warnAboutHardcodedVersionsIfAny(result.dependenciesWithHardcodedVersions)
         warnAboutDynamicVersionsIfAny(result.dependenciesWithDynamicVersions)
+        warnAboutGradleUpdateAvailableIfAny()
+    }
+
+    private fun warnAboutGradleUpdateAvailableIfAny() = runBlocking {
```

Comment:
This `runBlocking` call defeats the purpose of using coroutines for parallelization as it always delays the completion of the `refreshVersions` task, regardless of the connection quality. We can do better. I'll rework that before merging (hopefully tonight).

Reply:
good point, not sure how to do that

Reply:
I'm integrating Gradle version lookup into `VersionCandidatesLookupResult`. Looking for gradle updates will be performed by the existing `lookupVersionCandidates` that already parallelizes a bunch of work.

Reply:
Updated, and added support for Gradle snapshots (nightlies and release nightlies), and release candidates.

Here's an example taken from `sample-kotlin`:

```
The Gradle version used in this project is not up to date.
To update from version 6.5.1, run one of these commands:
./gradlew wrapper --gradle-version 6.6.1
./gradlew wrapper --gradle-version 6.7-rc-2

Be sure to read the migration guides to have a smooth upgrade process.
Note that you can replace with a specific intermediate version if needed.
```

If we were using a Gradle snapshot/nightly, the updates would show up here as well.

Reply:
Note: I dropped the `$` because not all shells start with a dollar (zsh is `%` for example), and it makes copy pasting easier.